make-up/code/5305-dea-bibovska/src/app/edit-recipie/edit-recipie.component.ts

```diff
@@ -0,0 +1,62 @@
+import { Component, } from '@angular/core';
+import { ActivatedRoute, Router } from '@angular/router';
+import { RecipieServiceService } from '../recipie-service.service';
+import { Recipie } from '../models/client';
+import { Observable, filter, map, switchMap } from 'rxjs';
+import { isNumeric } from '../common/utils';
+import { FormBuilder, FormGroup } from '@angular/forms';
+
+@Component({
+  selector: 'app-recipie-edit',
+  templateUrl: './recipie-edit.component.html',
+  styleUrls: ['./recipie-edit.component.css']
+})
+export class RecipieEditComponent {
+  recipie?: Recipie;
+  editForm: FormGroup = new FormGroup({});
+
+  constructor(
+    private route: ActivatedRoute,
+    private recipiesService: RecipieServiceService,
+    private router: Router,
+    private fb: FormBuilder,
+  ) { }
+
+  ngOnInit(): void {
+    this.route.paramMap.pipe(
+      map(params => params.get('id')),
+      filter(id => isNumeric(id)),
+      map(id => parseInt(id!, 10)),
+      switchMap(id => this.recipiesService.getRecipie(id)),
+    ).subscribe({
+      next: recipie => {
+        this.recipie = recipie;
+        this.prepareForm();
+      },
+      error: err => console.log(err)
+    });
+  }
+
+  prepareForm() {
+    this.editForm = this.fb.group({
```

Comment:
This code would have worked if it was connected to a corresponding markup, now, it's just idle code.